crmsh/bootstrap.py

```diff
@@ -180,12 +182,54 @@ def _validate_nodes_option(self):
         """
         Validate -N/--nodes option
         """
-        if not self.node_list:
+        if self.type == "join":
+            self.current_user = userdir.getuser()
+            if self.cluster_node.find('@') != -1:
+                user, node = self.cluster_node.split('@')
+                self.user_list = [user]
+                self.cluster_node = node
+            else:
+                self.user_list = [userdir.getuser()]
+            self.node_list = [self.cluster_node]
+
+            # TODO! This is too simple.
+            # When the another node joins, previous nodes will not know.
+            crm_conf_hosts_option = self.current_user + '@' + utils.this_node() + \
+                    ', ' + self.user_list[0] + '@' + self.cluster_node
+            config.set_option('core', 'hosts', crm_conf_hosts_option)
+            config.save()
             return
+
         self.node_list = utils.parse_append_action_argument(self.node_list)
         me = utils.this_node()
-        if me in self.node_list:
-            self.node_list.remove(me)
+        _node_list = []
+        _user_list = []
+        was_localhost_already = False
+        self.current_user = userdir.getuser()
+        for user_node in self.node_list:
+            if user_node.find('@') != -1:
+                user, node = user_node.split('@')
+            else:
+                node = user_node
+                user = userdir.getuser()
+            if node == me:
+                if was_localhost_already:
+                    utils.fatal("Duplicated input for -N/--nodes option")
+                was_localhost_already = True
+                if user != userdir.getuser():
+                    utils.fatal("Overriding current user '{}' by '{}'. Ouch, don't do it.".format(
+                        userdir.getuser(), user))
```

Comment:
Still, above `userdir.getuser()` should replaced as `self.current_user`

Reply:
I'll do it. _I didn't pay attention to this aspect of performance, you can find many other places where I call getuser() function_.